lib/logstash/outputs/logstash.rb

```diff
@@ -145,19 +149,21 @@ def ssl_trust_options
       elsif @ssl_certificate_authorities&.any?
         fail(LogStash::ConfigurationError, 'SSL Certificate Authorities cannot be configured when `ssl_verification_mode => none`') if @ssl_verification_mode == 'none'
 
-        trust_options['cacert'] = @ssl_certificate_authorities.first
+        trust_options['ssl_certificate_authorities'] = @ssl_certificate_authorities.first
       elsif @ssl_truststore_path
         fail(LogStash::ConfigurationError, 'SSL Truststore cannot be configured when `ssl_verification_mode => none`') if @ssl_verification_mode == 'none'
 
-        trust_options['truststore'] = @ssl_truststore_path
-        trust_options['truststore_type'] = keystore_type(@ssl_truststore_path)
-        trust_options['truststore_password'] = @ssl_truststore_password || fail("`truststore_password` is REQUIRED when `ssl_truststore_path` is provided")
+        trust_options['ssl_truststore_path'] = @ssl_truststore_path
+        trust_options['ssl_truststore_type'] = keystore_type(@ssl_truststore_path)
+        trust_options['ssl_truststore_password'] = @ssl_truststore_password || fail(LogStash::ConfigurationError, '`ssl_truststore_password` is REQUIRED when `ssl_truststore_path` is provided')
+      elsif @ssl_truststore_password
+        fail(LogStash::ConfigurationError, '`ssl_truststore_password` not allowed unless `ssl_truststore_path` is configured')
       end
     end
   end
 
   def keystore_type(keystore_path)
-    return 'JKS' if keystore_path.downcase.end_with?('.jks')
```

Comment:
with capital letters, we get an invalid setting error, comes through `ssl_keystore_type`/`ssl_truststore_type` of [http mixin client](https://github.com/logstash-plugins/logstash-mixin-http_client/blob/main/lib/logstash/plugin_mixins/http_client.rb):
```
[2023-09-14T12:30:21,218][ERROR][logstash.outputs.http    ] Invalid setting for http output plugin:

  output {
    http {
      # This setting must be a ["pkcs12", "jks"]
      # Expected one of ["pkcs12", "jks"], got ["JKS"]
      ssl_keystore_type => "JKS"
      ...
    }
  }
```